autosubmit/notifications/mail_notifier.py

```diff
@@ -187,39 +196,50 @@ def notify_experiment_status(
                 if temp_dir:
                     temp_dir.cleanup()
 
-        for mail in mail_to:
-            message['To'] = email.utils.formataddr((mail, mail))
-            try:
-                self._send_mail(self.config.MAIL_FROM, mail, message)
-            except BaseException as e:
-                Log.printlog(
-                    f'Trace:{str(e)}\nAn error has occurred while sending a mail for '
-                    f'warn about remote_platform', 6011)
+        formatted_addresses = [
+            email.utils.formataddr(
+                (mail, mail)) for mail in mail_to]
+        message["To"] = ", ".join(formatted_addresses)
+        try:
+            self._send_mail(
+                self.config.MAIL_FROM,
+                formatted_addresses,
+                message)
+        except Exception as e:
+            Log.printlog(
+                f'Trace:{str(e)}\nAn error has occurred while sending a warning mail '
+                f'about remote_platform', 6011)
 
     def notify_status_change(
             self,
             exp_id: str,
             job_name: str,
             prev_status: str,
             status: str,
-            mail_to: List[str]) -> None:
-        if not isinstance(mail_to, list):
-            raise ValueError('mail_to must be a list of emails!')
+            mail_to: list[str]) -> None:
+
+        _check_mail_address(mail_to)
         message_text = _generate_message_text(
             exp_id, job_name, prev_status, status)
         message = MIMEText(message_text)
         message['From'] = email.utils.formataddr(
             ('Autosubmit', self.config.MAIL_FROM))
         message['Subject'] = f'[Autosubmit] The job {job_name} status has changed to {status}'
         message['Date'] = email.utils.formatdate(localtime=True)
-        for mail in mail_to:  # expects a list
-            message['To'] = email.utils.formataddr((mail, mail))
-            try:
-                self._send_mail(self.config.MAIL_FROM, mail, message)
-            except BaseException as e:
-                Log.printlog(
-                    f'Trace:{str(e)}\nAn error has occurred while sending a mail '
-                    f'for the job {job_name}', 6011)
+
+        formatted_addresses = [
+            email.utils.formataddr(
+                (mail, mail)) for mail in mail_to]
+        message["To"] = ", ".join(formatted_addresses)
+        try:
+            self._send_mail(
+                self.config.MAIL_FROM,
+                formatted_addresses,
+                message)
+        except Exception as e:
+            Log.printlog(
+                f'Trace:{str(e)}\nAn error has occurred while sending a warning mail '
+                f'about remote_platform', 6011)
```

Comment:
IDE warned about code duplication here, and I think this could also become a function, similar to what you did with `_check_mail_address`

![image](https://github.com/user-attachments/assets/916a1336-d395-4ea0-bd25-a3cc34e4f1ed)